lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/YarnSubmissionHelper.java

```diff
@@ -201,10 +202,33 @@ public YarnSubmissionHelper setQueue(final String queueName) {
     return this;
   }
 
+  /**
+   * Sets the launcher class for the job.
+   * @param launcherClass
+   * @return
+   */
+  public YarnSubmissionHelper setLauncherClass(final Class launcherClass) {
+    this.launcherClazz = launcherClass;
+    return this;
+  }
+
+  /**
+   * Sets the configuration file for the job.
+   * Note that this does not have to be the Driver TANG configuration. In the bootstrap
+   * launch case, this can be the Avro file that supports the generation of a driver
+   * configuration file natively at the Launcher.
+   * @param configurationFileName
+   * @return
+   */
+  public YarnSubmissionHelper setConfigurationFileName(final String configurationFileName) {
+    this.confFileName = configurationFileName;
+    return this;
+  }
+
   public void submit() throws IOException, YarnException {
     // SET EXEC COMMAND
-    final List<String> launchCommand = new JavaLaunchCommandBuilder(commandPrefixList)
-        .setConfigurationFileName(this.fileNames.getDriverConfigurationPath())
+    final List<String> launchCommand = new JavaLaunchCommandBuilder(launcherClazz, commandPrefixList)
```

Comment:
Because it's a required parameter, and I didn't want this particular required parameter to be "hidden" with {{set}}. The parameter should only be set once anyway.